presto-main/src/main/java/io/prestosql/operator/scalar/ArrayContainsSubarray.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator.scalar;
+
+import io.airlift.slice.Slice;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.function.Convention;
+import io.prestosql.spi.function.Description;
+import io.prestosql.spi.function.OperatorDependency;
+import io.prestosql.spi.function.ScalarFunction;
+import io.prestosql.spi.function.SqlNullable;
+import io.prestosql.spi.function.SqlType;
+import io.prestosql.spi.function.TypeParameter;
+import io.prestosql.spi.type.StandardTypes;
+import io.prestosql.spi.type.Type;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.prestosql.spi.function.InvocationConvention.InvocationArgumentConvention.BLOCK_POSITION;
+import static io.prestosql.spi.function.InvocationConvention.InvocationArgumentConvention.NEVER_NULL;
+import static io.prestosql.spi.function.InvocationConvention.InvocationReturnConvention.NULLABLE_RETURN;
+import static io.prestosql.spi.function.OperatorType.EQUAL;
+
+@Description("Determines whether given values exist in the array")
```

Comment:
This does not seem to be correct.   I believe this function is determining if the array contains all of the values of the other array.  So this is more of a `contains_all` function.  For a `contains_subarray`, I would assume that the first array must contain the second array, with the values in the exact order... so a [String-searching algorithm](https://en.wikipedia.org/wiki/String-searching_algorithm), but with generic types instead of just characters.  Specifically, I would expect:
```
select contains_subarray(array['1','2','3','4'], array['2',3'])  -> true
select contains_subarray(array['1','2','3','4'], array['1',3'])  -> false
```
If you really want contains all, this can be done today with `all_match(array['2',3'], e -> contains(array['1','2','3','4'], e))`

Reply:
Thanks for your detailed explanation, I mixed the two functions up, `contains_subarray` should determine whether an array contains another, with the values in the exact order